starknet_devnet/blueprints/rpc.py

```diff
@@ -0,0 +1,681 @@
+            receipt = await get_transaction_receipt(transaction["txn_hash"])
+            combined = {**receipt, **transaction}
+            transactions_and_receipts.append(combined)
+        return transactions_and_receipts
+
+    block_number = block.block_number
+    old_root = state.starknet_wrapper.blocks.get_by_number(
+        block_number=block_number - 1) if block_number - 1 >= 0 else "0x0"
+
+    mapping: dict[str, Callable] = {
+        "TXN_HASH": transaction_hashes,
+        "FULL_TXNS": transactions,
+        "FULL_TXN_AND_RECEIPTS": full_transactions,
+    }
+    transactions: list = await mapping[requested_scope]()
+
+    block: RpcBlock = {
+        "block_hash": rpc_felt(block.block_hash),
+        "parent_hash": rpc_felt(block.parent_block_hash) or "0x0",
+        "block_number": block.block_number if block.block_number is not None else 0,
+        "status": block.status.name,
+        "sequencer": hex(DEFAULT_GENERAL_CONFIG.sequencer_address),
+        "new_root": str(block.state_root),
+        "old_root": str(old_root),
+        "accepted_time": block.timestamp,
+        "transactions": transactions,
+    }
+    return block
+
+
+class RpcStorageDiff(TypedDict):
+    """
+    TypedDict for rpc storage diff
+    """
+    address: str
+    key: str
+    value: str
+
+
+class RpcContractDiff(TypedDict):
+    """
+    TypedDict for rpc contract diff
+    """
+    address: str
+    contract_hash: str
+
+
+class RpcStateDiff(TypedDict):
+    """
+    TypedDict for roc state diff
+    """
+    storage_diffs: List[RpcStorageDiff]
+    contracts: List[RpcContractDiff]
+
+
+class RpcStateUpdate(TypedDict):
+    """
+    TypedDict for rpc state update
+    """
+    block_hash: str
+    new_root: str
+    old_root: str
+    accepted_time: int
+    state_diff: RpcStateDiff
+
+
+def rpc_state_update(state_update: dict) -> RpcStateUpdate:
+    """
+    Convert gateway state update to rpc state update
+    """
+    def storage_diffs() -> List[RpcStorageDiff]:
+        _storage_diffs = []
+        for address, storage in state_update["state_diff"]["storage_diffs"].items():
+            diff: RpcStorageDiff = {
+                "address": address,
+                "key": storage["key"],
+                "value": storage["value"]
+            }
+            _storage_diffs.append(diff)
+        return _storage_diffs
+
+    def contracts() -> List[RpcContractDiff]:
+        _contracts = []
+        for contract in state_update["state_diff"]["deployed_contracts"]:
+            diff: RpcContractDiff = {
+                "address": contract["address"],
+                "contract_hash": contract["contract_hash"]
+            }
+            _contracts.append(diff)
+        return _contracts
+
+    rpc_state: RpcStateUpdate = {
+        "block_hash": state_update["block_hash"],
+        "new_root": state_update["new_root"],
+        "old_root": state_update["old_root"],
+        "accepted_time": 0, # FIXME remove hardcoded value
```

Comment:
Gateway state update doesn't have an accepted time at all so I'm not sure how to replace that

Reply:
Is taking the block timestamp an option?

Reply:
I think we could do that. Since gateway state_update doesn't have a accepted time, this would require additional call to `get_block`, but that's probably not a big deal.

Reply:
Done